src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -81,30 +82,49 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     apiOperations
   )
 
-  import kantan.csv.generic._
-  import tech.cryptonomic.conseil.util.ConfigUtil.Csv._
+  implicit val tokens = initAnyCsvConfig()
 
-  /** Inits registered tokens at startup */
-  implicit val tokenContracts: TokenContracts = {
+  /* Reads csv resources to initialize db tables and smart contracts objects */
+  def initAnyCsvConfig(): TokenContracts = {
+    import kantan.csv.generic._
+    import tech.cryptonomic.conseil.util.ConfigUtil.Csv._
 
-    val futureTokenContracts =
-      TezosDb.initTableFromCsv(db, Tables.RegisteredTokens, tezosConf.network, separator = '|').map {
-        case (tokenRows, _) =>
-          TokenContracts.fromTokens(
-            tokenRows.map {
-              case Tables.RegisteredTokensRow(_, tokenName, standard, accountId) =>
-                ContractId(accountId) -> standard
-            }
-          )
+    val tokenContracts: TokenContracts = {
 
-      }
+      val tokenContractsFuture =
+        TezosDb.initTableFromCsv(db, Tables.RegisteredTokens, tezosConf.network, separator = '|').map {
+          case (tokenRows, _) =>
+            TokenContracts.fromConfig(
+              tokenRows.map {
+                case Tables.RegisteredTokensRow(_, tokenName, standard, accountId) =>
+                  ContractId(accountId) -> standard
+              }
+            )
+
+        }
 
-    Await.result(futureTokenContracts, 5.seconds)
+      Await.result(tokenContractsFuture, 5.seconds)
+    }
+
+    /** Inits tables with values from CSV files */
+    TezosDb.initTableFromCsv(db, Tables.KnownAddresses, tezosConf.network)
+    TezosDb.initTableFromCsv(db, Tables.BakerRegistry, tezosConf.network)
+
+    //return the contracts definitions
+    tokenContracts
   }
 
-  /** Inits tables with values from CSV files */
-  TezosDb.initTableFromCsv(db, Tables.KnownAddresses, tezosConf.network)
-  TezosDb.initTableFromCsv(db, Tables.BakerRegistry, tezosConf.network)
+  implicit val tns: TNSContract =
+    tezosConf.tns match {
```

Comment:
I have not been through the code changes yet but should there not be a change in reference.conf?

Reply:
the `tns` sub-tree of the configuration is optional, and it's custom-tailored for each network, depending on the contract address.

Should I add a placeholder here with fake/empty data?

Reply:
I think as a rule all configuration entries should show up in reference.conf as it's a self-documenting reference for all users.

Reply:
done